src/SMatrix.jl

```diff
@@ -66,7 +66,7 @@ end
         SMatrix{S1, S2, $T, L}(x)
     end
 end
-typealias SMatrixNoType{S1, S2, L, T} SMatrix{S1, S2, T, L}
+@compat SMatrixNoType{S1, S2, L, T} = SMatrix{S1, S2, T, L}
```

Comment:
I don't know why, but I don't remember this existing, which is why I couldn't fathom the test failure... In theory, `MMatrix` should be symmetric. But don't worry about this for this PR.